cli/command_test.go

```diff
@@ -273,29 +274,112 @@ func TestRootCommand_Run(t *testing.T) {
 	}
 }
 
-func TestBaseCommand_Prompt_Values(t *testing.T) {
+//nolint:thelper // These aren't actually helpers, and we want the failures to show the correct line
+func TestBaseCommand_Prompt(t *testing.T) {
 	t.Parallel()
 
 	ctx := logging.WithLogger(context.Background(), logging.TestLogger(t))
 
 	cases := []struct {
-		name      string
-		args      []string
-		msg       string
-		inputs    []string
-		err       string
-		expStderr string
+		name    string
+		execute func(t *testing.T, cmd *BaseCommand)
 	}{
 		{
-			name:   "sets_input_value",
-			args:   []string{"prompt"},
-			msg:    "enter input value:",
-			inputs: []string{"input"},
-		}, {
-			name:   "handles_multiple_prompts",
-			args:   []string{"prompt"},
-			msg:    "enter input value:",
-			inputs: []string{"input1", "input2"},
+			name: "reads_line",
+			execute: func(t *testing.T, cmd *BaseCommand) {
```

Comment:
I'd suggest that it makes more sense for these to be separate test rather than defining these closures. There's not much benefit that I see to combining these three subtest into a single test.

Reply:
I've generally always followed the belief that all tests for a logical unit (in this case a function) should reside under `Test<Thing>`, and use subtests to group the logic and output into something reasonable. That way, when looking at test output, it's immediately clear which struct, function, and subtest is failing. I don't have to spelunk around multiple test functions to figure out where a failure is coming fromt.

I can break them apart into their own `t.Run()` statements if you think that'd be useful, but I don't want to create `TestPrompt_reads_line`, `TestPrompt_reads_multiple`, etc. I find that extremely difficult to parse.

The disadvantage of moving away from the subtest is that we have to duplicate the cli setup logic. If that were ever refactored, it would need to be updated in many places instead of just once.

Reply:
OK, I'm fine with that.